src/osd/OSD.h

```diff
@@ -1631,9 +1632,8 @@ class OSD : public Dispatcher,
       }
     }; // struct ShardData
 
-    vector<ShardData*> shard_list;
     OSD *osd;
-    uint32_t num_shards;
+    ceph::tiny_vector<ShardData, 8> shards;
```

Comment:
So this is now limited to 8 shards max, right? Surely we can set it at runtime rather than compile-time?
(I haven't looked through the tiny_vector in any detail yet, just wanted to see the invocation.)

Reply:
Nope, arbitrary number of shards, set at run-time, can be handled. The template parameter specifies only the size of preallocated, continuous storage `tiny_vector` has inside to avoid dynamic allocations -- this is what makes it different from Boost's `static_vector`.

I personally perceive `tiny_vector` as `small_vector` for handling weird things like `std::mutexes`, `std::atomics` and so on.